nncf/experimental/onnx/algorithms/min_max_quantization.py

```diff
@@ -0,0 +1,163 @@
+            if quantizer_config.per_channel:
+                axes = (0, 2, 3)
+                return ONNXMinMaxStatisticCollector(is_symmetric, axes)
+            return ONNXMinMaxStatisticCollector(is_symmetric, None)
+
+    def _create_model_transformer(self, model: onnx.ModelProto):
+        self.model_transformer = ONNXModelTransformer(model)
+
+    def _get_quantizer_setup(self, model: onnx.ModelProto):
+        nncf_graph = GraphConverter.create_nncf_graph(model)
+        ip_graph = InsertionPointGraph(nncf_graph)
+        pattern = ONNX_HW_FUSED_PATTERNS.get_full_pattern_graph()
+        ip_graph = ip_graph.get_ip_graph_with_merged_hw_optimized_operations(pattern)
+
+        weight_nodes = nncf_graph.get_nodes_by_metatypes(QUANTIZATION_LAYER_METATYPES)
+        quantizable_layer_nodes = [QuantizableWeightedLayerNode(weight_node, [QuantizerConfig()]) for weight_node in
+                                   weight_nodes]
+
+        hw_config_type = HWConfigType.from_str(HW_CONFIG_TYPE_TARGET_DEVICE_MAP[self.target_device])
+        hw_config_path = ONNXHWConfig.get_path_to_hw_config(hw_config_type)
+        hw_config = ONNXHWConfig.from_json(hw_config_path)
+
+        solver = QuantizerPropagationSolver(ignored_scopes=self.ignored_scopes,
+                                            hw_config=hw_config,
+                                            default_trait_to_metatype_map=DEFAULT_ONNX_QUANT_TRAIT_TO_OP_DICT,
+                                            default_qconfig_list=[self._get_default_qconfig()],
+                                            quantizable_layer_nodes=quantizable_layer_nodes,
+                                            quantize_outputs=self.quantize_outputs)
+
+        quantization_proposal = solver.run_on_ip_graph(ip_graph)
+        multi_config_setup = quantization_proposal.quantizer_setup
+        single_config_setup = multi_config_setup.select_first_qconfig_for_each_point()
+        finalized_proposal = quantization_proposal.finalize(single_config_setup)
+        final_setup = solver.get_final_quantizer_setup(finalized_proposal)
+        return final_setup
+
+    def apply(self, model: onnx.ModelProto, engine: ONNXEngine):
+        modified_model = modify_onnx_model_for_quantization(model)
+        self._create_model_transformer(modified_model)
+        transformation_layout = ONNXTransformationLayout()
+        transformation_commands = []
+        onnx_graph = ONNXGraph(modified_model)
+
+        # If statistics were not collected
+        if not self.statistics_collector.layers_statistics:
+            layers_to_collect_statistics = self.get_layers_for_statistics(modified_model)
+            self.statistics_collector.register_layer_statistics(layers_to_collect_statistics)
+            self.statistics_collector.collect_statistics(modified_model)
+        layers_statistics = self.statistics_collector.layers_statistics
```

Comment:
Is there any possibility to get only needed statistics for each algorithm?

Reply:
I see your point. 
I saw that at POT for that reason collected statistics represents in the format - `Dict[Algorithm: statistics]`.
I think it could be done in the same manner.

Reply:
I mean that we should optimize the `for` loop below and don't walk through whole `layers_statistics`. Maybe something like `self.statistics_collector.algo_name.layers_statistics` should be there, or something like that.

Reply:
Optimized. 
Now we iterate through only activations layers, which obtained from the algo and then find this layers in dictionary with the statistics.